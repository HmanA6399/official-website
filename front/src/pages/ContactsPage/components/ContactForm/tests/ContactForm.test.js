import React from "react";
import ContactForm from "../index";
import testCases from "./test_cases.json";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

// Globals
let nameInput, emailInput, messageInput;

// Mocking window.scrollTo to hide the error
// src: https://qiita.com/akameco/items/0edfdae02507204b24c8
const noop = () => {};
Object.defineProperty(window, 'scrollTo', { value: noop, writable: true });

beforeAll(() => {
  const { contactForm, getByTestId } = render(<ContactForm />);
  nameInput = getByTestId("nameInput");
  emailInput = getByTestId("emailInput");
  messageInput = getByTestId("messageInput");
});

test("ContactsInfo renders successfuly", () => {
  const componentTree = renderer.create(<ContactForm />);
  expect(componentTree).toMatchSnapshot();
});

test("Input is entered to the textboxes correctly", () => {
  nameInput.value = testCases[0].name;
  emailInput.value = testCases[0].email;
  messageInput.value = testCases[0].message;
  expect(nameInput.value).toBe(testCases[0].name);
  expect(emailInput.value).toBe(testCases[0].email);
  expect(messageInput.value).toBe(testCases[0].message);
});


