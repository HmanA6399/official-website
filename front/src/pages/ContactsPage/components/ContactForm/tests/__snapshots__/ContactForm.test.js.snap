// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsInfo renders successfuly 1`] = `
<article
  className="bg-section"
  id="contact-form"
>
  <h2
    className="section-title text-center"
  >
    Leave a message
  </h2>
  <form
    noValidate={true}
    onChange={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="flex-row-2"
    >
      <div
        className="box"
      >
        <label
          htmlFor="name"
        >
          Name
           
          <span
            className="sup-span"
          >
            <sup>
              *
            </sup>
          </span>
        </label>
        <input
          data-testid="nameInput"
          id="name"
          name="name"
          type="text"
        />
        
      </div>
      <div
        className="box"
      >
        <label
          htmlFor="email"
        >
          Email
           
          <span
            className="sup-span"
          >
            <sup>
              *
            </sup>
          </span>
        </label>
        <input
          data-testid="emailInput"
          id="email"
          name="email"
          type="text"
        />
        
      </div>
      <div
        className="box"
      >
        <label
          htmlFor="message"
        >
          Your Message
           
          <span
            className="sup-span"
          >
            <sup>
              *
            </sup>
          </span>
        </label>
        <textarea
          data-testid="messageInput"
          id="message"
          name="message"
        />
        
      </div>
    </div>
    <input
      type="submit"
      value="SEND"
    />
  </form>
</article>
`;
